junifer/preprocess/bold_warper.py

```diff
@@ -110,46 +137,110 @@ def preprocess(
-            # Replace original BOLD data with warped BOLD data
-            _, input = apply_warper.preprocess(
-                input=input,
-                extra_input=extra_input,
-            )
-        elif warp_file_ext == ".h5":
-            logger.debug("Using ANTs with BOLDWarper")
-            # Initialize AntsApplyTransformsWarper for computation
-            ants_apply_transforms_warper = _AntsApplyTransformsWarper(
-                reference=self.ref, on="BOLD"
-            )
-            # Replace original BOLD data with warped BOLD data
-            _, input = ants_apply_transforms_warper.preprocess(
-                input=input,
-                extra_input=extra_input,
-            )
+        logger.info(f"Warping BOLD to {self.ref} space using BOLDWarper")
+        # Transform to native space
+        if self.ref == "T1w":
+            # Check for extra inputs
+            if extra_input is None:
+                raise_error(
+                    "No extra input provided, requires `Warp` and "
+                    f"`{self.ref}` data types in particular."
+                )
+            # Check for warp file type to use correct tool
+            warp_file_ext = extra_input["Warp"]["path"].suffix
+            if warp_file_ext == ".mat":
+                logger.debug("Using FSL with BOLDWarper")
+                # Initialize ApplyWarper for computation
+                apply_warper = _ApplyWarper(reference=self.ref, on="BOLD")
+                # Replace original BOLD data with warped BOLD data
+                _, input = apply_warper.preprocess(
+                    input=input,
+                    extra_input=extra_input,
+                )
+            elif warp_file_ext == ".h5":
+                logger.debug("Using ANTs with BOLDWarper")
+                # Initialize AntsApplyTransformsWarper for computation
+                ants_apply_transforms_warper = _AntsApplyTransformsWarper(
+                    reference=self.ref, on="BOLD"
+                )
+                # Replace original BOLD data with warped BOLD data
+                _, input = ants_apply_transforms_warper.preprocess(
+                    input=input,
+                    extra_input=extra_input,
+                )
+            else:
+                raise_error(
+                    msg=(
+                        "Unknown warp / transformation file extension: "
+                        f"{warp_file_ext}"
+                    ),
+                    klass=RuntimeError,
+                )
+        # Transform to template space
         else:
-            raise_error(
-                msg=(
-                    "Unknown warp / transformation file extension: "
-                    f"{warp_file_ext}"
-                ),
-                klass=RuntimeError,
-            )
+            # Check pre-requirements for space manipulation
+            if self.ref == input["space"]:
+                logger.info(
```

Comment:
This should be a warning if not an error. Why is the BOLDWarper set if you don't need to warp? I prefer an error though.

Reply:
I don't understand why it should fail with an error, can make it a warning though. But I prefer a logged message instead of a warning in these cases.